sys/include/net/coap.h

```diff
@@ -157,20 +157,51 @@ extern "C" {
 /** @} */
 
 /**
- * @name    Timing parameters
+ * @defgroup net_coap_conf    CoAP compile configurations
+ * @ingroup  config
  * @{
  */
+/**
+ * @brief    Timeout in seconds for a response to a confirmable request
+ *
+ * This value is for the response to the *initial* confirmable message. The
+ * timeout doubles for subsequent retries. To avoid synchronization of resends
+ * across hosts, the actual timeout is chosen randomly between COAP_ACK_TIMEOUT
+ * and (COAP_ACK_TIMEOUT * COAP_RANDOM_FACTOR).
+ */
+#ifndef COAP_ACK_TIMEOUT
 #define COAP_ACK_TIMEOUT        (2U)
+#endif
+
+/** @brief   Used to calculate upper bound for timeout; see COAP_ACK_TIMEOUT */
+#ifndef COAP_RANDOM_FACTOR
 #define COAP_RANDOM_FACTOR      (1.5)
+#endif
+
 /**
- * @brief   Maximum variation for confirmable timeout.
+ * @brief   Approximation for maximum variation for confirmable timeout
  *
  * Must be an integer, defined as:
  *
  *     (COAP_ACK_TIMEOUT * COAP_RANDOM_FACTOR) - COAP_ACK_TIMEOUT
+ *
+ * Like COAP_ACK_TIMEOUT, this value is valid for the initial confirmable
+ * message, and doubles for subsequent retries.
  */
+#ifndef COAP_ACK_VARIANCE
 #define COAP_ACK_VARIANCE       (1U)
+#endif
+
+/** @brief   Maximum number of retransmissions for a confirmable request */
+#ifndef COAP_MAX_RETRANSMIT
 #define COAP_MAX_RETRANSMIT     (4)
+#endif
+/** @} */
+
+/**
+ * @name    Timing parameters
```

Comment:
Aren't the once above not also timing parameters still?

Reply:
Yes, they are. I can't find a way to use the same name "Timing parameters" to keep the list of timing parameters that also are compile configurations _separate_ from those timing parameters  that are not also compile configurations.

If I use a different name for the timing parameters that also are compile configurations, like "Timing parameters compile configurations", then the lists stay separate. I'd appreciate any suggestions on the best approach.

Reply:
They could stay separate, but if the others are also timing parameters, maybe "Fixed timing parameters"?

Reply:
Thanks for the suggestion. I created a separate commit for that.